packages/myst-frontmatter/src/utils/socialLinks.ts

```diff
@@ -34,35 +30,59 @@ export const SOCIAL_LINKS_ALIASES = {
   instagram: 'threads', // This is the same username
 };
 
+export type SocialLinks = {
+  [key in typeof SOCIAL_LINKS_KEYS[number]]?: string;
+}
+
 export function validateSocialLinks(
   input: any,
   opts: ValidationOptions,
-  output: SocialLinks = {},
+  output?: SocialLinks,
 ): SocialLinks | undefined {
-  const value = output
+  const value: SocialLinks = output
     ? input
-    : validateObjectKeys(input, { optional: SOCIAL_LINKS_KEYS, alias: SOCIAL_LINKS_ALIASES }, opts);
+    : validateObjectKeys(input, { optional: SOCIAL_LINKS_KEYS as unknown as string[], alias: SOCIAL_LINKS_ALIASES }, opts);
+
+  if (value === undefined) return undefined;
+
+  const result = output ?? {};
 
+  // FIXME: normalize usernames into URLs
   if (defined(value.url)) {
-    output.url = validateUrl(value.url, incrementOptions('url', opts));
+    result.url = validateUrl(value.url, incrementOptions('url', opts));
   }
   if (defined(value.github)) {
-    output.github = validateString(value.github, incrementOptions('github', opts));
+    result.github = validateString(value.github, incrementOptions('github', opts));
   }
   if (defined(value.bluesky)) {
-    output.bluesky = validateString(value.bluesky, incrementOptions('bluesky', opts));
+    result.bluesky = validateString(value.bluesky, incrementOptions('bluesky', opts));
   }
   if (defined(value.mastodon)) {
-    output.mastodon = validateString(value.mastodon, incrementOptions('mastodon', opts));
+    result.mastodon = validateString(value.mastodon, incrementOptions('mastodon', opts));
   }
   if (defined(value.linkedin)) {
-    output.linkedin = validateUrl(value.linkedin, incrementOptions('linkedin', opts));
+    result.linkedin = validateUrl(value.linkedin, incrementOptions('linkedin', opts));
   }
   if (defined(value.threads)) {
-    output.threads = validateString(value.threads, incrementOptions('threads', opts));
+    result.threads = validateString(value.threads, incrementOptions('threads', opts));
   }
   if (defined(value.twitter)) {
-    output.twitter = validateString(value.twitter, incrementOptions('twitter', opts));
+    result.twitter = validateString(value.twitter, incrementOptions('twitter', opts));
```

Comment:
`validateString` has a regex option so if we want to be a bit more strict, we could do something like:

```suggestion
    result.twitter = validateString(value.twitter, {
      ...incrementOptions('twitter', opts),
      regex: '^@',
    });
```

Reply:
I do think we should probably take a stance on if these are usernames or urls, and normalize between them. That could come in a different PR though.

I could make arguments for either direction (URL or username). Once we make a choice and a validator, that logic could be shared in the theme for backwards compatibility.

I would prefer if we don't _require_ starting with `@` — makes the yaml harder to write!

Reply:
When I last spoke to @kne42 about this, I'd suggested accepting either usernames or URLs for most social fields. I looked at this PR again to try and move it forward, and I feel more inclined to be a bit more restrictive here and accept (in some cases) only usernames:

1. Parsing URLs is fiddly, and there are some cases that are nontrivial — Mastodon doesn't guarantee that `foo.com/@bar` is a valid URL. 
3. Mapping from URL to username sometimes requires a fetch or is not well defined (mastodon)

Reply:
I've pushed a commit that adds validators for mastodon (only `@username@domain`) and bsky (`@domain` or `https://bsky.app/profile/domain`)! I'll pick this up tomorrow to visit the rest.